### PR TITLE
Add getNextSortingOrder to sorting column api

### PR DESCRIPTION
**Pull request by mr-bjerre**
Add getNextSortingOrder to resolve the discussion in #3944

### CODE REVIEW

**Comment by mr-bjerre** [comment]
I'd say `column.getNextSortingOrder()` would be even better than the more low level method `column.getSortDescFirst()`. The only use case for `column.getSortDescFirst()` is to determine `column.getNextSortingOrder()` AFAIK.

But `toggleSorting` seems pretty complex so I'm not sure how to make `getNextSortingOrder`.

Why this in the first place? You might want to hint the next sorting order on mouse hover. E.g. see https://tailwindui.com/components/application-ui/lists/tables#component-7b5a46e74e475708d966ca31716f1771

File: packages/table-core/src/features/Sorting.ts [hunk 1]
```diff
@@ -278,6 +279,24 @@ export const Sorting: TableFeature = {
...
+        const firstSortDirection = sortDescFirst ? 'desc' : 'asc'
+
+        const isSorted = column.getIsSorted()
+        if (!isSorted) {
+          return firstSortDirection
+        }
+        if (isSorted === firstSortDirection) {
+          return isSorted === 'desc' ? 'asc' : 'desc'
+        } else {
+          return 'none'
```
Comment (tannerlinsley) [inline packages/table-core/src/features/Sorting.ts:296]: Let’s make this a false, that way we can check it for truthiness much easier.

**Review by tannerlinsley — CHANGES_REQUESTED** [review]
Any way we can have the toggler logic use this new function?

**Comment by mr-bjerre** [comment]
Yeah I was going in that direction first but then my head started spinning in the toggler logic 😂

I'll revisit again tomorrow (Copenhagen time here so done for the night)

File: packages/table-core/src/features/Sorting.ts [hunk 2]
```diff
@@ -191,6 +192,9 @@ export const Sorting: TableFeature = {
         //   return
         // }
 
+        // this needs to be outside of instance.setSorting to be in sync with rerender
+        const nextSortingOrder = column.getNextSortingOrder()
```
Comment (mr-bjerre) [inline packages/table-core/src/features/Sorting.ts:196]: @tannerlinsley I first put this line inside `instance.setSorting` but then the react sorting example broke when `sortAction` became `toggle`. It wasn't clear to me exactly why but I assume some state not being in sync on rerender.

**Comment by mr-bjerre** [comment]
Updated the PR - let me know if it needs more attention

**Comment by mr-bjerre** [comment]
Maybe I should change the method so we return both sortAction and the next sortDirection (if any) instead?

**Review by tannerlinsley — APPROVED** [review]

**Pull request merged**